### PR TITLE
ADU support for STM32L4 and SAM E54

### PR DESCRIPTION
**Pull request by nik-markovic**
Also includes size optimizations for SAM E54

### CODE REVIEW

File: iotc-azrtos-sdk/include/iotconnect_certs.h [hunk 1]
```diff
@@ -35,6 +35,9 @@ extern const unsigned int IOTCONNECT_BALTIMORE_ROOT_CERT_SIZE;
 extern const unsigned char IOTCONNECT_GODADDY_G2_ROOT_CERT[];
 extern const unsigned int IOTCONNECT_GODADDY_G2_ROOT_CERT_SIZE;
 
+/* Digicert Global Root G2*/
+extern const unsigned char DIGICERT_GLOBAL_ROOT_G2[];
+extern const unsigned int DIGICERT_GLOBAL_ROOT_G2_SIZE;
```
Comment (nik-markovic) [inline iotc-azrtos-sdk/include/iotconnect_certs.h]: rename DIGICERT_GLOBAL_ROOT_G2* to IOTCONNECT_DIGICERT_GLOBAL_ROOT_G2*

**Review by nmatthews-witekio — COMMENTED** [review]
Thanks for merging my setup-project.sh change -- and apologies it conflicts with your's here.
Looks like the wget you modified should now be ${FETCH}.

Appreciate this isn't code you've changed:
  Looks like potential double free on line 178 of iotconnect_app.c
  (url is freed also on line 202 -- suggest delete line 178).
  
In samples/stm32l4/sensors-demo/src/main.c
Add: 
#include "sensors_data.h"
so that init_sensors() is declared.

Remove:
static void sample_thread_entry(ULONG parameter);
this is not mentioned in rest of code.

Add:
#if 0 / #endif pair around:
```
static NX_SNTP_CLIENT   sntp_client;
  
/* System clock time for UTC.  */
static ULONG            unix_time_base;
```
They are unused but mentioned in a later #if 0 block

Ideally, a more descriptive name than "0" would be good to tie these two elements together -- or move code inside the existing #if 0 block.

There seems to be a lot of added trailing whitespace, not sure if that's intentional

I wonder if iotc-azrtos-sdk/src/iotconnect_certs.c would be good to have some mechanism to select particular certificates -- my immediate thought was a #ifdef around each certificate/length (but there has to be a better way than that). 

The new IOTCONNECT_DIGICERT_GLOBAL_ROOT_G2 will potentially add 914 bytes to all code that links in iotc-azrtos-sdk/src/iotconnect_certs.c -- unless the linker is capable.

I would be tempted to make arrays static -- maybe with an accessor function -- but maybe that's getting too far out of scope.

-----

P.S. And completely unrelated:
I noticed some PNGs in the repo -- maybe pngcrush would make them smaller -- but would have to check that boards are still ok with them.

**Comment by nik-markovic** [comment]
@nmatthews-witekio
I applied changes from main regarding unzip/fetch.

Good catch on that double-free. I think you or someone pointed it out earlier, but maybe I missed fixing it.

For #if zeros, there's also a minor change in the "[Fix double-free](https://github.com/avnet-iotconnect/iotc-azurertos-sdk/pull/40/commits/e12d8b65c37fadab4b1a5e9036d73aaa0372eed5)" commit, in addition to [Add some comments to explain that we are removing code with #if 0](https://github.com/avnet-iotconnect/iotc-azurertos-sdk/pull/40/commits/bbf4b33bf136b024290d3df57bf970e533918050)

Regarding iotc-azrtos-sdk/src/iotconnect_certs.c, most compilars should remove unused certs uless someone forgot section-related compiler and linker flags. You can try this yourself - add a new dummy cert array and then recompile the stm32l4 project. The .bin file size should be the same. Obviously, make sure that it's only declared and not actually used anywhere.

Also, DiciCert root CA cert we are about to [start using it](https://techcommunity.microsoft.com/t5/internet-of-things-blog/azure-iot-tls-critical-changes-are-almost-here-and-why-you/ba-p/2393169) not just for OTA (unless it's a different one)

This trailing whitespace, there could be some and I don't think that any of it should be intentional. I don't even know how to easily spot those. I mean a few extra bytes on files shouldn't really matter.

Which PNGs? These sizes seem reasonable. 
```shell
 du -sh ./samples/stm32u5/media/ ./samples/wfi32iot/media/
320K	./samples/stm32u5/media/
2.8M	./samples/wfi32iot/media/
```
I don't see a point in making them smaller.

I'd say accessor fnction for constants is overkill. Maybe if you have a reason that I am not seeing. Naked globals (as **variables**, not constants), I would always try avoid. AzureRTOS code seem to have a lot of global variables. If there's any in our code it's probably beause we inherited it.

**Comment by nmatthews-witekio** [comment]
Ok, I only raised about the added cert as that looked like it would affect all boards, not just STM32L4 and SAM E54 -- so that would be a deal more testing?

I've no strong objections if you need to merge :-)

**Comment by nik-markovic** [comment]
> Ok, I only raised about the added cert as that looked like it would affect all boards, not just STM32L4 and SAM E54 -- so that would be a deal more testing?

All of the certs are needed down the road anyways - both Baltimore DigiCert and GoDaddy all actually linked into the code and being in use. We just haven't come around to it - waiting on some infro from Softwebe.

I don't plan on testing anything because of extra 1k aded to the project IN THE CASE that optimization flags are "bugged". We can optimize those projects as we find the need for it.

What I am trying to say is that we got, for example, from Microsoft a project that was 48% of flash used. I turned on those flags and got to 25%. That's value added. We can add value if we want, but we don't need to unless we are really running out of Flash/RAM.

**Review by nmatthews-witekio — APPROVED** [review]
Looks fine to me.

**Pull request merged**